modules/account-lib/src/coin/sol/solInstructionFactory.ts

```diff
@@ -98,16 +99,40 @@ function memoInstruction(data: Memo): TransactionInstruction[] {
  */
 function transferInstruction(data: Transfer): TransactionInstruction[] {
   const {
-    params: { fromAddress, toAddress, amount },
+    params: { fromAddress, toAddress, amount, mint, source },
   } = data;
   assert(fromAddress, 'Missing fromAddress param');
   assert(toAddress, 'Missing toAddress param');
   assert(amount, 'Missing toAddress param');
-  const transferInstruction = SystemProgram.transfer({
-    fromPubkey: new PublicKey(fromAddress),
-    toPubkey: new PublicKey(toAddress),
-    lamports: new BigNumber(amount).toNumber(),
-  });
+  let transferInstruction;
+  if (!mint) {
+    // native transfer
+    transferInstruction = SystemProgram.transfer({
+      fromPubkey: new PublicKey(fromAddress),
+      toPubkey: new PublicKey(toAddress),
+      lamports: new BigNumber(amount).toNumber(),
+    });
+  } else {
+    // token transfer
+    assert(mint, 'Missing mint');
+    assert(source, 'Missing source');
+    let decs;
+    coins.forEach((value, key) => {
+      if (value['tokenAddress'] === mint) {
+        decs = value.decimalPlaces;
+      }
+    });
+    transferInstruction = splToken.Token.createTransferCheckedInstruction(
+      splToken.TOKEN_PROGRAM_ID,
+      new PublicKey(source),
+      new PublicKey(mint),
+      new PublicKey(toAddress),
+      new PublicKey(fromAddress),
+      [],
+      new BigNumber(amount).toNumber(),
+      decs,
+    );
```

Comment:
what's the difference between `source` and `fromAddress`?

Reply:
my mistake, fromAddress IS the source address - fixed this by keeping the sender (ata addr for tokens) called fromAddress, to be compatable with native transfers, but using an optional owner field in Transfer (instead of fromAddress & source, as proper use of this would be duplicate fields)